apps/public-docsite-v9/src/DocsComponents/FluentDocsPage.stories.tsx

```diff
@@ -133,30 +144,108 @@ const getNativeElementsList = (elements: SBEnumType['value']): JSX.Element => {
   );
 };
 
-const RenderArgsTable = ({ hideArgsTable, primaryStory }: { primaryStory: PrimaryStory; hideArgsTable: boolean }) => {
+const slotRegex = /as\?:\s*"([^"]+)"/;
+
+function withSlotEnhancer(story: PreparedStory) {
+  const hasArgAsProp = story.argTypes.as && story.argTypes.as?.type?.name === 'enum';
```

Comment:
to accomodate suggestion regarding simplifing logic, these is needed

```suggestion
  const hasArgAsProp = story.argTypes.as?.type?.name === 'enum';
  const argAsProp = hasArgAsProp ? (story.argTypes.as.type as SBEnumType).value : null;
```